src/matches.rs

```diff
@@ -112,34 +112,40 @@ impl LateLintPass for MatchPass {
 fn check_single_match(cx: &LateContext, ex: &Expr, arms: &[Arm], expr: &Expr) {
     if arms.len() == 2 &&
        arms[0].pats.len() == 1 && arms[0].guard.is_none() &&
-       arms[1].pats.len() == 1 && arms[1].guard.is_none() &&
-       is_unit_expr(&arms[1].body) {
+       arms[1].pats.len() == 1 && arms[1].guard.is_none() {
```

Comment:
Yeah, I'm also not sure about this – on one hand the `match` statement has one more level of nesting, which may lead to more line breaks which hurt readability, on the other hand `if let` will often lead to one more line of *very* sparse code, which takes more vertical space then necessary.